src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -281,6 +282,39 @@ public void InitializeService(ServiceHost serviceHost, SqlToolsContext context)
 
         #region Request Handlers
 
+        /// <summary>
+        /// T-SQL syntax parse request callback
+        /// </summary>
+        /// <param name="param"></param>
+        /// <param name="requestContext"></param>
+        /// <returns></returns>
+        internal async Task HandleSyntaxParseRequest(SyntaxParseParams param, RequestContext<SyntaxParseResult> requestContext)
+        {            
+            try
+            {            
+                ParseResult result = Parser.Parse(param.Query);
```

Comment:
should the entire request handler be wrapped in a Task to avoid blocking the dispatcher queue?  Or are we sure the `Parse` will be quick.  Since it's disconnected parse it probably will be quick, but may be safe to run on different thread.